packages/internal/src/natural-sort.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @param {string} a
+ * @param {string} b
+ * @returns {-1 | 0 | 1}
+ */
+const compareNats = (a, b) => {
+  // Default to IEEE 754 number arithmetic for speed, but fall back on bigint
+  // arithmetic to resolve ties because big numbers can lose resolution,
+  // sometimes even becoming infinite.
+  const diff = +a - +b;
+  const finiteDiff =
+    (Number.isFinite(diff) && diff) || Number(BigInt(a) - BigInt(b));
```

Comment:
Should we special case `a === b` to avoid BigInt conversions ?

Reply:
I don't think so; the BigInt fallback is present for correctness but we shouldn't actually have need to natural-sort values in which embedded numbers exceed 2**53—and if/when we do, re-measurement of relative performance will be warranted anyway.

Reply:
Yeah, seems worth it:
```console
$ scripts/esbench.mjs -b1 -h 'V8,*XS*' \
  --init '
    const R = max => Math.floor(Math.random() * max);
    const makeArray = (length, makeElement) => Array.from({ length }, makeElement);
    Object.assign(globalThis, {
      diff_nocheckeq: (a, b) => {
        return (+a - +b) || Math.sign(Number(BigInt(a) - BigInt(b)));
      },
      diff_checkeq: (a, b) => {
        return (+a - +b) || (a === b ? 0 : Math.sign(Number(BigInt(a) - BigInt(b))));
      },
      inputs: makeArray(1000, (_, i) => String(RegExp(`[1-9][0-9]{${i % 5}}`).exec(Math.random()) || 1)),                                                                                                   });                   
  ' \                                                             
  nocheckeq:'for(const x of inputs) result = diff_nocheckeq(x, x)' \                                
  checkeq:'for(const x of inputs) result = diff_checkeq(x, x)'
#### Moddable XS
nocheckeq (0) 0.34479371316306484 ops/ms after 27 13-count samples
checkeq (0) 0.845771144278607 ops/ms after 50 17-count samples

#### V8
nocheckeq (0) 3.225 ops/ms after 215 15-count samples
checkeq (0) 4.62 ops/ms after 154 30-count samples
```